### PR TITLE
Fix resolveTask

### LINKED ISSUE #991: Using "recently used" tasks launches the wrong task
This might be an issue with my local setup as I've been messing around with task settings a lot but it would help if someone else could test if they are getting the same issue or not. 

Essentially if I run the "R: document" task and then re-run it by selecting the "recently used" tasks it launches the wrong task (the build task instead).  I've added a video below to highlight the issue:

https://user-images.githubusercontent.com/8447209/153753282-f81a5c93-69f7-4d2f-81ae-22913380be84.mp4

If the issue isn't repeatable feel free to close as it probably means I've just borked my local config :)

### PR DESCRIPTION
**Pull request by renkun-ken**
# What problem did you solve?

Closes #991

### CODE REVIEW

File: src/tasks.ts [hunk 1]
```diff
@@ -2,106 +2,98 @@
 
 import * as vscode from 'vscode';
 
+interface RTaskDefinition extends vscode.TaskDefinition {
```
[the remark this answers is not in the document]
Comment (ManuelHentschel) [reply src/tasks.ts:5]: Just my personal opinion/experience: Last time I tried (~1 year ago), the task api was quite confusing to work with and not really flexible. I tried implementing the typical build/test tasks with the possibility to supply optional arguments to e.g. `devtools::install()`, but this was not supported by the task API. Instead, the extension had to explicitly register all possible combinations of arguments, which is not feasible. Apparently, the idea is to scan a list of tasks specified elsewhere, e.g. package.json, and parse them.

[the remark this answers is not in the document]
Comment (renkun-ken) [reply src/tasks.ts:5]: [TaskDefinition](https://code.visualstudio.com/api/extension-guides/task-provider#task-definition) is basically the schema of tasks user could specify in `tasks.json` for that particular task type, in our case, `R`.

Previously, our `resolveTask` does not really resolve the task.

**Comment by renkun-ken** [comment]
Currently, there is a problem that #964 uses `ShellQuotedString` in the arguments supplied to `ShellExecution`. Then clicking the icon on the right of the task label 

image

will result in an invalid task definition:

```json
{
	"type": "R",
	"command": "Rscript",
	"args": [
		"-e",
		{
			"value": "devtools::install()",
			"quoting": 2
		}
	],
	"problemMatcher": [],
	"label": "R: Install"
}
```

I'm wondering if it is necessary to use `ShellExecution` in the first place? If we use `ProcessExecution` instead, then there seems to be no quoting issue at all?

**Comment by renkun-ken** [comment]
I tested using `ProcessExecution` on Windows, Linux and macOS, and the pre-defined tasks work as expected. And user does not need to take care of argument quoting when they write tasks like this:

```r
{
	"version": "2.0.0",
	"tasks": [
		{
			"type": "R",
			"command": "Rscript",
			"args": [
				"-e",
				"readLines('${file}')",
			],
			"cwd": "${workspaceRoot}",
			"env": {
				"ZZZ_TEST": "test1"
			},
			"problemMatcher": [],
			"label": "R: print file"
		}
	]
}
````

**Comment by renkun-ken** [comment]
However, in Windows, the file paths are supplied as is. Maybe we could use some form of notation in `args` so that the file paths could be properly supplied as R string literals.

```
> Executing task: C:\Program Files\R\R-4.1.2\bin\Rscript.exe -e readLines('$c:\Users\ken\Documents\languageserver\.vscode\tasks.json') <

Error: '\U' used without hex digits in character string starting "'$c:\U"
Execution halted
The terminal process "C:\Program Files\R\R-4.1.2\bin\Rscript.exe '-e', 'readLines('$c:\Users\ken\Documents\languageserver\.vscode\tasks.json')'" terminated with exit code: 1.
```

**Comment by renkun-ken** [comment]
Thanks @gowerc for the nice work done in https://github.com/renkun-ken/vscode-R/pull/3.

**Comment by renkun-ken** [comment]
Windows user could use raw string in task code to make use of the path variables:

```json
{
	"version": "2.0.0",
	"tasks": [
		{
			"type": "R",
			"code": [
				"readLines(r'{${file}}')"
			],
			"group": "build",
			"problemMatcher": [],
			"label": "R test"
		}
	]
}
```

**Comment by gowerc** [comment]
@renkun-ken 

I was wondering if we should document the use of tasks in the readme? Personally I never found the current way vscode exposes them to users to be that intuitive so was thinking we might get more adoption if we had some more docs on it?

**Comment by renkun-ken** [comment]
Yes, of course. I'll add an entry to [Features](https://github.com/REditorSupport/vscode-R#features) which points to [Tasks](https://github.com/REditorSupport/vscode-R/wiki/Tasks) in wiki.

**Comment by gowerc** [comment]
Makes sense to me, will try and add something

**Pull request merged**